pennylane/drawer/utils.py

```diff
@@ -149,8 +153,13 @@ def cwire_connections(layers, bit_map):
             classical conditions or measurement statistics as keys.
 
     Returns:
-        list, list: list of list of accessed layers for each classical wire, and largest wire
-        corresponding to the accessed layers in the list above.
+        dict, dict, dict: The first dictionary is the updated ``bit_map``, potentially with
+        some mid-circuit measurements mapped to new (smaller) classical wires. The second and third
+        dictionaries have the classical wires as keys and lists of lists as values, with the outer
+        list running over different (re)usages of the classical wire. For the second dictionary,
+        the inner lists contain the indices of the accessed layers, for the third dictionary,
+        they contain the measured quantum wires and the largest quantum wire of conditionally
+        applied operations (no entries for terminal statistics of mid-circuit measurements).
```

Comment:
So for the `connected_layers` and `connected_wires`, will the keys ever not be successive integers? AKA will we ever skip a classical wire, or have a classical wire that's not an integer? What does dictionaries get us that lists don't?

Reply:
I think it's mostly to avoid a list of list of lists. In principle, it also allows a more flexible handling of classical wires if we ever want to move away from consecutive integers.
So in my head, this is just the right data structure, which happens to be used in a way that a list of list of lists could also handle, but it wouldn't generalize. I suppose we get fast access on top, although that should not matter (?)